docs/proposal/2022-11-22-botkube-plugins-integration.md

```diff
@@ -0,0 +1,542 @@
+			 botkube/kubectl@v1.2.0:  # not allowed as it is conflicting with already registered 'kubectl' from the Botkube repository in different version.
+				 enabled: true
+				 config:
+					 commands:
+						 verbs: ["get","logs","top"]
+	 ```
+   </details>
+
+3. We cannot have bindings to the same executor but from different repositories or versions:
+   <details>
+     <summary>Example</summary>
+
+   ```yaml
+   communications:
+     default-group:
+       socketSlack:
+         enabled: true
+         channels:
+           default:
+             name: botkubers
+             bindings:
+               executors:            # such binding is not allowed as it has conflicting 'kubectl' executor from different repositories
+                 - kubectl-read-only
+                 - kubectl-pods-rw
+   executors:
+     'kubectl-read-only':
+       botkube/kubectl:
+         enabled: true
+         config:
+           # ...
+     'kubectl-pods-rw':
+       mszostok/kubectl:
+         enabled: true
+         config:
+           # ...
+     'kubectl-deploy-rw':
+       botkube/kubectl@v1.0.0:
+         enabled: true
+         config:
+           # ...
+   ```
+   </details>
+
+#### Plugin configuration
+
+The plugin specific configuration is not validated by Botkube. It's plugins responsibility to do that. In the future, we can allow specifying JSON Schema for each plugin defined in the index file, for example:
+
+```yaml
+entries:
+  - name: "kubectl"
+    type: "executor"
+    description: "Kubectl executor plugin."
+    urls:
+      - https://github.com/kubeshop/botkube/releases/download/v0.17.0/executor_kubectl-darwin-amd64
+    jsonSchema:
+      value: |-
+        {
+          "$schema": "http://json-schema.org/draft-07/schema",
+          "type": "object",
+          "required": [
+              "version"
+          ],
+          "properties": {
+            "version": {
+              "$id": "#/properties/version",
+              "type": "string",
+              "minLength": 5,
+              "pattern": "^(?:0|[1-9]\\d*)\\.(?:0|[1-9]\\d*)\\.(?:0|[1-9]\\d*)$",
+              "title": "Kubernetes version",
+              "description": "Kubernetes version",
+              "default": "",
+              "examples": [
+                  "1.19.0"
+              ]
+            }
+          },
+          "additionalProperties": true
+        }
+```
+
+This will be used by Botkube to validate that configuration defined by user is valid before even starting a given plugin.
+
+### Passing configuration to a given plugin
+
+Current Botkube implementation allows you to specify different executor and sources configuration and bind them to a single channel. Later they are merged together based on the given business logic.
+
+To support the same experience for plugins, we need to pass a list of configuration each time we run the:
+
+- `Execute` method for executor plugins
+- `Source` method for source plugins
```

Comment:
It would be good to clarify when these methods are run. I think this will be accurate:

```suggestion
- `Execute` method for executor plugins, invoked when user runs a given command,
- `Source` method for source plugins, invoked during Botkube startup.
```